ebos/eclproblem.hh

```diff
@@ -3020,115 +3020,112 @@ private:
+                        break;
+                    case FaceDir::XPlus:
+                        data = &massratebcX_;
+                        break;
+                    case FaceDir::YMinus:
+                        data = &massratebcYMinus_;
+                        break;
+                    case FaceDir::YPlus:
+                        data = &massratebcY_;
+                        break;
+                    case FaceDir::ZMinus:
+                        data = &massratebcZMinus_;
+                        break;
+                    case FaceDir::ZPlus:
+                        data = &massratebcZ_;
+                        break;
                     }
-                    else
-                        throw std::logic_error("invalid component name for BC");
-
-                    int i1 = bc.getRecord(record).getItem("I1").template get< int >(0) - 1;
-                    int i2 = bc.getRecord(record).getItem("I2").template get< int >(0) - 1;
-                    int j1 = bc.getRecord(record).getItem("J1").template get< int >(0) - 1;
-                    int j2 = bc.getRecord(record).getItem("J2").template get< int >(0) - 1;
-                    int k1 = bc.getRecord(record).getItem("K1").template get< int >(0) - 1;
-                    int k2 = bc.getRecord(record).getItem("K2").template get< int >(0) - 1;
-                    std::string direction = bc.getRecord(record).getItem("DIRECTION").getTrimmedString(0);
-
-                    if (type == "RATE") {
-                        assert(compIdx >= 0);
-                        std::vector<RateVector>* data = 0;
-                        if (direction == "X-")
-                            data = &massratebcXMinus_;
-                        else if (direction == "X")
-                            data = &massratebcX_;
-                        else if (direction == "Y-")
-                            data = &massratebcYMinus_;
-                        else if (direction == "Y")
-                            data = &massratebcY_;
-                        else if (direction == "Z-")
-                            data = &massratebcZMinus_;
-                        else if (direction == "Z")
-                            data = &massratebcZ_;
-                        else
-                            throw std::logic_error("invalid direction for BC");
-
-                        const Evaluation rate = bc.getRecord(record).getItem("RATE").getSIDouble(0);
-                        for (int i = i1; i <= i2; ++i) {
-                            for (int j = j1; j <= j2; ++j) {
-                                for (int k = k1; k <= k2; ++k) {
-                                    std::array<int, 3> tmp = {i,j,k};
-                                    size_t elemIdx = cartesianToCompressedElemIdx[vanguard.cartesianIndex(tmp)];
-                                    (*data)[elemIdx][compIdx] = rate;
-                                }
+
+                    const Evaluation rate = bcface.rate;
```

Comment:
Yes it could - it is plain double though.

Reply:
>> This could probably be a reference rather than an object.
>
> Yes it could - it is plain double though.

Sorry, my bad.  The original code intended type conversion here (`Scalar` to `Evaluation`).  I've gotten so used to writing (and reading)
```C++
const auto rate = Evaluation{ bcface.rate };
```
in this case that I didn't think of the implicit conversion behaviour.